src/manifolds/Sphere.jl

```diff
@@ -147,112 +214,113 @@ Represent the tangent vector `X` at point `p` from the [`Sphere`](@ref) `M` in
 an orthonormal basis by rotating the vector `X` using the rotation matrix
 $2\frac{q q^\mathrm{T}}{q^\mathrm{T} q} - I$ where $q = p + (1, 0, …, 0)$.
 """
-function get_coordinates(M::Sphere{N}, p, X, B::DefaultOrthonormalBasis) where {N}
-    if isapprox(p[1], 1)
+function get_coordinates(M::Sphere{n,ℝ}, p, X, B::DefaultOrthonormalBasis) where {n}
+    if isapprox(abs(p[1]), 1)
         return X[2:end]
     else
-        xp1 = p .+ ntuple(i -> ifelse(i == 1, 1, 0), N + 1)
-        return (2*xp1*dot(xp1, X)/dot(xp1, xp1)-X)[2:end]
+        xp1 = p .+ ntuple(i -> ifelse(i == 1, 1, 0), n+1)
+        return (2*xp1*real(dot(xp1, X))/real(dot(xp1, xp1))-X)[2:end]
```

Comment:
```suggestion
        return (2*xp1*dot(xp1, X)/dot(xp1, xp1)-X)[2:end]
```
This real does nothing since the method is only for the real sphere anyway.

Reply:
yes this is also a leftover from my try to get a complex sphere basis, that I missed to remove, thanks for noticing.